### ISSUE TITLE
Change Amp scoring to PID

### ISSUE THREAD

**Issue by noahlinton** [frc4039/2024#70]
Sometimes, the NOTE gets stuck in the roller and not fully ejected. 

Also, did some digging and found a difference between amp shooting and speaker shooting commands. Speaker uses shooterPID() method, but amp shooterSpeedControl() method. Any reason for the difference here? Should they both use PID? Could this be affecting the amp scoring?

**Comment by Ben-Alderson** [comment]
`shooterSpeedControl` allows independent control of top and bottom rollers. I also don't think that `shooterPID` will work correctly when we're putting so much more load on the motors to amp.

You probably just need to raise the shooter or indexer speed to fix the amp scoring.

**Comment by Ben-Alderson** [comment]
If we want to use a PID for the amp scoring it will need to be a separate PID tuned to react to that sustained load.

**Comment by noahlinton** [comment]
We investigated this a little today on REVERB on the field. We have looked into the code and into the telemetry via advantage kit. Here is what we have found.

1) We are controlling shooter wheels via percent output for amping as discussed above. (See next point) Based on telemetry, this gets us to about 2000 RPM. The threshold requirement for the indexer to start for amp scoring is only 500 RPM. So we are able to introduce the game piece before the shooter is up to speed.

2) We found that the scoring is sensitive to RPM. Maybe a PID controller on RPM here (set to 2000RPM for now) would help to keep things consistent? When we changed battery, performance increased.

3) we should change the threshold on the indexer to be closer to the desired set speed (2000RPM for now)

4) most of these values are hard coded, but we should probably clean up to be constants.

OPTIONAL FANCY CODE (we may need to reconsider this) 5) Instead of the indexer being "while held" for the driver right trigger, can we just run the indexer until we dont see the beam break anymore + we see the RPM of the upper wheel return to normal speed? (The RPM dips as the NOTE passes through the mechanism). This would remove some human error of stopping the indexer early.
